src/core/inductive.ml

```diff
@@ -55,17 +56,74 @@ let gen_ind_typ_codom : popt -> sym -> (tbox list -> tbox) -> string -> tbox =
     | _ -> fatal pos "The type of %a is not supported" pp_symbol ind_sym
   in aux [] !(ind_sym.sym_type)
 
+module T =
+  struct
+
+    (** [fold_app a [b1;...;bn]] returns (... ((a b1) b2) ...) bn. *)
+    let fold_app  : tbox -> tbox list -> tbox = List.fold_left _Appl
+
+    (** [fold_app_special f ts] returns the same result that
+        fold_app (_Symb [f]) [ts]. *)
+    let fold_app_special : sym -> tbox list -> tbox = fun f ts ->
+      fold_app (_Symb f) ts
+
+    (** [prf_of p c ts t] returns a proof of the form
+        c.symb_prf ( (((p ts1) ...) tsn) t) where ts = [ts1;...;tsn]. *)
+    let prf_of : tvar -> config -> tbox list -> tbox -> tbox = fun p c ts t ->
+      fold_app_special c.symb_prf [_Appl (fold_app (_Vari p) ts) t]
+  end
+
+(** [preprocessing pos c ind_typ_list p_str x_str] creates an associated list
+    between an inductive type and its predicat, and the list of conclusions,
+    according to the list of the type of the inductive types [ind_typ_list].
+    The strings [p_str] and [x_str] are useful to avoid name clashes.
+*)
+let preprocessing :
+      popt -> config -> sym list -> string -> string
+      -> ((sym * (tvar * tbox)) list * tbox list) =
+  fun pos c ind_typ_list p_str x_str ->
+  let prop = _Symb c.symb_Prop in
+
+  let rec aux :
+    int -> (sym * (tvar * tbox)) list -> tbox list -> sym list
+    -> ((sym * (tvar * tbox)) list * tbox list) =
+    fun i assoc_predicats conclusion_list ind_typ_list ->
+    match ind_typ_list with
+    | [] -> assoc_predicats, conclusion_list
+    | ind_sym::q ->
+        (* STEP 1 - Create the predicat (name + type) associated to the
+           symbol [ind_sym]. *)
+          (* A - Create the name of the predicat. *)
+        let p_str = p_str ^ (string_of_int i) in
```

Comment:
parentheses are useless here; thanks to remove them

Reply:
Why not using `"p_" ^ ind_sym.sym_name` for the names of predicate variables?

Reply:
> Why not using `"p_" ^ ind_sym.sym_name` for the names of predicate variables?

Because you prefer short names, hence number instead of the names of predicate variables.
It's as you want but please don't change your mind each month.